### PR TITLE
Updated event_name function to use kwargs

### PR DESCRIPTION
**Pull request by jonotassia**
Implemented the below changes to `event_name` (per #3852):

- [x] Update function
- [x] Update stubs
- [x] Update docs
- [x] Update and pass test case

If all looks well, I'll go ahead and start making the changes to the other functions as well!

### CODE REVIEW

**Comment by illume** [comment]
I see this error:
```
   Creating library build\temp.win32-3.6\Release\src_c\key.cp36-win32.lib and object build\temp.win32-3.6\Release\src_c\key.cp36-win32.exp
Generating code
src_c/event.c(2261): error C2220: warning treated as error - no 'object' file generated
src_c/event.c(2261): warning C4113: 'PyObject *(__cdecl *)(PyObject *,PyObject *,PyObject *)' differs in parameter lists from 'PyCFunction'
```
https://ci.appveyor.com/project/pygame/pygame/builds/47082115/job/1qdc339ah2q3eupw#L625


Seems there's a typo on the 'args' one is missing an s:
```
static PyObject *
event_name(PyObject *self, PyObject *args, PyObject *kwargs)
```

File: buildconfig/stubs/pygame/event.pyi [hunk 1]
```diff
@@ -37,7 +37,7 @@ def poll() -> Event: ...
 def wait(timeout: int = 0) -> Event: ...
 def peek(eventtype: Optional[_EventTypes] = None, pump: Any = True) -> bool: ...
 def clear(eventtype: Optional[_EventTypes] = None, pump: Any = True) -> None: ...
-def event_name(type: int) -> str: ...
+def event_name(type: int, **kwargs: Any) -> str: ...
```
[the remark this answers is not in the document]
Comment (jonotassia) [reply buildconfig/stubs/pygame/event.pyi:40]: I've changed this back now!

**Comment by jonotassia** [comment]
Thanks for catching that. I updated arg to args, but also did a little research, which indicated that in order to accept keyword args, I needed to cast the PyMethodDef for `event_name` as a PyCFunction, which I've done now as well:

`{"event_name", (PyCFunction)event_name, METH_VARARGS | METH_KEYWORDS, DOC_PYGAMEEVENTEVENTNAME},`

Hopefully that runs without any compiler errors this time.

**Review by illume — APPROVED** [review]
Thanks @jonotassia 🎉

**Pull request merged**